src/ephemeris.c

```diff
@@ -68,6 +70,129 @@ static s8 calc_sat_state_xyz(const ephemeris_t *e, const gps_time_t *t,
   return 0;
 }
 
+/** Re-calculation of ephemeris within the interval of measurement
+ *  ICD 5.1: A.3.1.2, with corrections from RTCM 3.2 p.186
+ *
+ * \param ydot Pointer to output array
+ * \param pos Pointer to position input array
+ * \param vel Pointer to velocity input array
+ * \param acc Pointer to acceleration input array
+ * \param e Pointer to GLO ephemeris
+ */
+static void calc_ydot(double ydot[6],
+                      const double pos[3],
+                      const double vel[3],
+                      const double acc[3])
+{
+
+  double r = sqrt(pow(pos[0], 2) +
+                  pow(pos[1], 2) +
+                  pow(pos[2], 2));
+
+  ydot[0] = vel[0];
+  ydot[1] = vel[1];
+  ydot[2] = vel[2];
+
+  ydot[3] = -GPS_GM / pow(r,3) * pos[0]
+            - 3/2 * GLO_J02 * GPS_GM * pow(GLO_A_E, 2) / pow(r, 5) *
+              pos[0] * (1 - 5 * pow(pos[2], 2) / pow(r, 2))
+            + pow(GPS_OMEGAE_DOT, 2) * pos[0]
+            + 2 * GPS_OMEGAE_DOT * vel[1]
+            + acc[0];
+
+  ydot[4] = -GPS_GM / pow(r,3) * pos[1]
+            - 3/2 * GLO_J02 * GPS_GM * pow(GLO_A_E, 2) / pow(r, 5) *
+              pos[1] * (1 - 5 * pow(pos[2], 2) / pow(r, 2))
+            + pow(GPS_OMEGAE_DOT, 2) * pos[1]
+            + 2 * GPS_OMEGAE_DOT * vel[0]
```

Comment:
Yeah so instead of ` + 2 * GPS_OMEGAE_DOT * vel[0]` it should be ` - 2 * GPS_OMEGAE_DOT * vel[0]`

If you look at the RTCM document or the Matlab code you will see line 100 should be +2 coefficient, and 107 a -2 coeffcient